local-plans/migrate-to-development-plan-specification.py

```diff
@@ -42,18 +42,28 @@ def hash_(value):
 
 
 # replace old form of national park identifiers with their digital land version
-# e.g. government-organisation:OT543 ==> national-park-authority:Q72617890 
-def create_organisation_id_mapper():
-    mapper = {}
-    f = open("./patch/national-park-authority.csv")
+# e.g. government-organisation:OT543 ==> national-park-authority:Q72617890
+def load_mappings(mapper, path, old_field, new_field, update_prefix=True):
+    f = open(path)
     reader = csv.DictReader(f)
     for row in reader:
-        if row.get("government-organisation") and row.get("national-park-authority"):
-            mapper[f"government-organisation:{row['government-organisation']}"] = f"national-park-authority:{row['national-park-authority']}" 
+        if row.get(old_field) and row.get(new_field):
+            if update_prefix:
+                mapper[f"{old_field}:{row[old_field]}"] = f"{new_field}:{row[new_field]}"
+            else:
+                mapper[f"{new_field}:{row[old_field]}"] = f"{new_field}:{row[new_field]}"
     return mapper
 
-id_mapper = create_organisation_id_mapper()
 
+def create_organisation_id_mapper():
+    mapper = {}
+    # load national-parks
+    mapper = load_mappings(mapper, "./patch/national-park-authority.csv", "government-organisation", "national-park-authority")
```

Comment:
I find this a bit odd because it both passes `mapper` in and also assigns it. I think I'd prefer a more functional approach of not passing and mutating `mapper` inside `load_mappings` function, but instead loading into a new data structure and returning it (results can be merged into a single dict by the caller). Either that, or could just remove the `mapper =` from both lines and rely on pythons "feature" of passing dicts by reference :)

Reply:
I think commit ff1517d is more along these lines